### ISSUE TITLE
Fix our weird posteriors

### ISSUE THREAD

**Issue by Justinezgh** [Ciela-Institute/CosmOrford#23]
Let's keep track here :) what needs to be tested:

- [x] To validate the compressor: regression plots @noe-dia 
- [x] To validate the NPE: fit the two mons @Justinezgh 
- [x] Normalization of the NF condition @JuliaLinhart

**Comment by Justinezgh** [comment]
NF is working: 
<img width="862" height="257" alt="Image" src="https://github.com/user-attachments/assets/f73d611b-0407-4142-b163-bddfc47729bd" />

**Comment by noe-dia** [comment]
Compressor is suspicious. Im going to double check if I have a normalization problem for the theta predicted vs true theta 

The plot is done for the compressor trained on the full challenge dataset (no pretraining) and the predictions are computed on the holdout dataset (I'm taking only 50 examples here). 

The notebook for the plot is on the `noe/compressor_check` branch, [here](https://github.com/Ciela-Institute/CosmOrford/blob/noe/compressor_check/notebooks/checking_compressor.ipynb)

<img width="1269" height="490" alt="Image" src="https://github.com/user-attachments/assets/9c52b4d7-8a9c-4320-9299-d7f76d196f2f" />


Just as a reference, here are the results for predictions computed on the training dataset. It seems the training dataset is very OOD with 
the holdout... 

<img width="1248" height="461" alt="Image" src="https://github.com/user-attachments/assets/d630dd5d-114c-47e0-b4a6-99ee207e4809" />

**Comment by Justinezgh** [comment]
Thanks @noe-dia! Do you have the corresponding wandb run? It feels like its overfitting

**Comment by noe-dia** [comment]
Just synced it here https://wandb.ai/cosmostat/neurips-wl-challenge/runs/x03aa5pi

**Comment by Justinezgh** [comment]
Thanks! Okay something is going on with the holdout sims. We should run PQMass to compare the holdout dataset to the one we use for training

**Comment by noe-dia** [comment]
So if I remember correctly, doesn't the holdout dataset contain additional cosmologies + the initial conditions that are not the same across kappa maps (compared to the training dataset where initial conditions are fixed and there are "just" 101 cosmologies) ? If that's the case, that would explain the big difference in predictions for the compressor. 

I don't remember if there is a reason to keep the same setting as the challenge where the holdout dataset is OOD compared to the training dataset.

**Comment by JuliaLinhart** [comment]
So I guess there's just a big distribution shift for the kappa maps, holdout has bigger std and a larger value range:

```
train: kappa mean=-0.0001, std=0.0083, min=-0.1201, max=1.4141
holdout: kappa mean=-0.0001, std=0.0205, min=-0.1663, max=1.5479
```

Theta values seem to be the same between train and holdout datasets.

**Comment by Justinezgh** [comment]
@b-remy  would it be okay if we use the validation split of `CosmoStat/neurips-wl-challenge-flat` (5656 sims) for training the npe and remove one cosmology from the whole dataset that will be used to compute the FoM?

**Comment by JuliaLinhart** [comment]
Oh and for normalization, i do have results, but they look more or less the same as before. Given the issue comes from the compressor not the NF...

**Comment by b-remy** [comment]
@Justinezgh you can always use the validation sims to train the npe, it's just that it's a small set for training an npe, and you might need to to a validation split for this one as well.

I am going to look into why the compressor is failing at predicting cosmologies for the holdout set, while being good for flat set.

**Comment by b-remy** [comment]
Reading your notebook @noe-dia, I see that you're adding noise to the holdout kappa map. But these maps should be already noisy!  

The holdout dataset was given by the challenge organiser to test our inference method, so these maps are observed kappa maps, so noisy. 

Can you try without adding noise to the holdout maps?

**Comment by b-remy** [comment]
Can you check this for your npe contours @JuliaLinhart

**Comment by JuliaLinhart** [comment]
I was almost certain that it was going to be a preprocessing error! thx @b-remy i will check some time this weekend if i can!

**Comment by noe-dia** [comment]
Nice catch @b-remy ! Just did the same regression plot without adding the noise on the holdout dataset, and it makes much more sense now. I took the code from the `run_npe_budget_scan.py` script without changing it, so we need to fix it before running the NPE training again 

<img width="1098" height="404" alt="Image" src="https://github.com/user-attachments/assets/47081de3-6422-4096-8fbd-158c47815690" />

**Comment by noe-dia** [comment]
Question, do we have access to the holdout dataset without any noise ? In the npe code right now, we compute summaries for different noise realizations for each kappa map, but if the dataset is already noisy we only have one noise realization per kappa map.

I don't know how important that is

**Comment by b-remy** [comment]
Kappa maps in the holdout set are already noisy and masked, so we cannot do augmentation on them. We'll have to work with only one noise realization per kappa map for the nde, but I think that's ok

**Comment by b-remy** [comment]
And thanks for your tests and plots @JuliaLinhart @noe-dia @Justinezgh  !

**Comment by noe-dia** [comment]
Sounds good, I'll try to make a PR for @Justinezgh tomorrow with a fix of the `run_npe_budget_scan.py`

**Comment by Justinezgh** [comment]
Yes it might overfit more quickly, so maybe we will need to reduce the number of training steps but it should be fine

**Comment by noe-dia** [comment]
Just created the PR [here](https://github.com/Ciela-Institute/CosmOrford/pull/24)

**Comment by Justinezgh** [comment]
Fixed! See https://github.com/Ciela-Institute/CosmOrford/issues/10#issuecomment-4187128131
Thanks everyone :)
